From 55c384708fc5cc5ba028cfa2f223e4effb6367c6 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:38 -0400
Subject: [usb] xhci: extend quirk for Renesas cards

Message-id: <1410887463-20674-387-git-send-email-dzickus@redhat.com>
Patchwork-id: 95380
O-Subject: [RHEL7 PATCH 386/591] xhci: extend quirk for Renesas cards
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 6db249ebefc6bf5c39f35dfaacc046d8ad3ffd70
Author: Igor Gnatenko <i.gnatenko.brain@gmail.com>
Date:   Fri Apr 25 19:20:15 2014 +0300

    xhci: extend quirk for Renesas cards

    After suspend another Renesas PCI-X USB 3.0 card doesn't work.
    [root@fedora-20 ~]# lspci -vmnnd 1912:
    Device: 03:00.0
    Class: USB controller [0c03]
    Vendor: Renesas Technology Corp. [1912]
    Device: uPD720202 USB 3.0 Host Controller [0015]
    SVendor: Renesas Technology Corp. [1912]
    SDevice: uPD720202 USB 3.0 Host Controller [0015]
    Rev: 02
    ProgIf: 30

    This patch should be applied to stable kernel 3.14 that contain
    the commit 1aa9578c1a9450fb21501c4f549f5b1edb557e6d
    "xhci: Fix resume issues on Renesas chips in Samsung laptops"

    Reported-and-tested-by: Anatoly Kharchenko <rfr-bugs@yandex.ru>
    Reference: http://redmine.russianfedora.pro/issues/1315
    Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>
    Cc: stable@vger.kernel.org # 3.14
    Signed-off-by: Mathias Nyman <mathias.nyman@linux.intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
index ca830df..35d4477 100644
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -144,6 +144,9 @@ static void xhci_pci_quirks(struct device *dev, struct xhci_hcd *xhci)
     "QUIRK: Resetting on resume");
   xhci->quirks |= XHCI_TRUST_TX_LENGTH;
  }
+ if (pdev->vendor == PCI_VENDOR_ID_RENESAS &&
+   pdev->device == 0x0015)
+  xhci->quirks |= XHCI_RESET_ON_RESUME;
  if (pdev->vendor == PCI_VENDOR_ID_VIA)
   xhci->quirks |= XHCI_RESET_ON_RESUME;
 }
-- 
1.7.1